mexico-city/src/mc_nitro.rs

```diff
@@ -0,0 +1,219 @@
+            return Err(format!(
+                "nc-nitro::get_psa_attestation_token nsm_lib_init failed:{:?}",
+                nsm_fd
+            ));
+        }
+        let status = unsafe {
+            nsm_lib::nsm_get_attestation_doc(
+                nsm_fd,                                  //fd
+                enclave_cert_hash.as_ref().as_ptr() as *const u8,      // user_data
+                enclave_cert_hash.as_ref().len() as u32,               // user_data_len
+                challenge.as_ptr(),                      // nonce_data
+                challenge.len() as u32,                  // nonce_len
+                std::ptr::null() as *const u8,          // pub_key_data
+                0 as u32,                               // pub_key_len
+                att_doc.as_mut_ptr(),                    // att_doc_data
+                &mut att_doc_len,                        // att_doc_len
+            )
+        };
+        match status {
+            nsm_io::ErrorCode::Success => (),
+            _ => return Err(format!("nc-nitro::get_psa_attestation_token received non-success error code from nsm_lib:{:?}", status)),
+        }
+        unsafe {
+            att_doc.set_len(att_doc_len as usize);
+        }
+        att_doc.clone()
+    };
+    let enclave_name: String = managers::baja_manager::get_enclave_name()
+        .map_err(|err| format!("mc_nitro::nitro_main failed to get enclave name from baja_manager:{:?}", err))?;
+    let nre_message = NitroRootEnclaveMessage::ProxyAttestation(challenge.to_vec(), nitro_token, enclave_name);
+    let nre_message_buffer = bincode::serialize(&nre_message)
+        .map_err(|err| format!("mc_nitro::get_psa_attestation_token failed to serialize NRE message:{:?}", err))?;
+    
+    // send the buffer back to Sinaloa via an ocall
+    let vsocksocket = vsocket::vsock_connect(HOST_CID, OCALL_PORT)
+        .map_err(|err| format!("mc_nitro::get_psa_attestation_token vsock_connect failed:{:?}", err))?;
+    send_buffer(vsocksocket.as_raw_fd(), &nre_message_buffer)
+        .map_err(|err| format!("mc_nitro::get_psa_attestation_token send_buffer failed:{:?}", err))?;
+    let received_buffer = receive_buffer(vsocksocket.as_raw_fd())
+        .map_err(|err| format!("mc_nitro::get_psa_attestation_token receive_buffer failed:{:?}", err))?;        
+    let received_message: NitroRootEnclaveMessage = bincode::deserialize(&received_buffer)
+        .map_err(|err| format!("mc_nitro::get_psa_attestation_token deserialize failed:{:?}", err))?;
+
+    let (psa_token, pubkey, device_id) = match received_message {
+        NitroRootEnclaveMessage::PSAToken(token, pubkey, d_id) => (token, pubkey, d_id),
+        _ => return Err(format!("mc_nitro::get_psa_attestation_token received invalid message from NRE:{:?}", received_message)),
+    };
+    let psa_token_message: MCMessage = MCMessage::PSAAttestationToken(psa_token, pubkey, device_id.try_into().unwrap());
+
+    return Ok(psa_token_message);
```

Comment:
The `return` isn't needed here.

Reply:
I am aware. However, I have recently concluded that I find that explicitly putting `return` (instead of relying on the missing` ;` to indicate a return) much easier to read and debug.

This is non-idomatic rust, so I look forward to hearing yours and others opinion on this, as this is a new opinion of mine that I'm willing to reconsider.

Reply:
I don't particularly mind it, but as you say it isn't idiomatic Rust.  I'll mark this and all similar comments as resolved.